Builds with Clang on Arm64 are broken

### Godot version

f130ad622a

### System information

Ubuntu 22.04, Clang 14.0

### Issue description

Building Godot with Clang for Arm64 fails with:


```
$ scons platform=linuxbsd use_llvm=yes use_lld=yes
[...]
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:13:
In file included from thirdparty/embree/common/sys/../simd/arm/emulation.h:12:
thirdparty/embree/common/sys/../simd/arm/sse2neon.h:5957:33: error: cannot initialize a parameter of type 'float32x4_t' (vector of 4 'float32_t' values) with an lvalue of type '__m128d' (aka 'float64x2_t')
    __builtin_nontemporal_store(a, (float32x4_t *) p);
```

This relates to the PRs #69144 and #70108 which should have targeted GCC only.

### Steps to reproduce

Build using  an Arm64 machine with the following parameters:
`scons platform=linuxbsd use_llvm=yes use_lld=yes`

### Minimal reproduction project

N/A

--- a/modules/raycast/SCsub
+++ b/modules/raycast/SCsub
@@ -96,7 +96,7 @@ if env["builtin_embree"]:
 
     if not env.msvc:
         # Flags synced with upstream gnu.cmake.
-        if env["arch"] == "arm64" and env["platform"] == "linuxbsd":
+        if env["arch"] == "arm64" and env["platform"] == "linuxbsd" and not env["use_llvm"]:
             env_thirdparty.Append(CXXFLAGS=["-flax-vector-conversions"])
 
         env_thirdparty.Append(

